docs/proposals/container-runtime-interface/capabilities.md

```diff
@@ -0,0 +1,137 @@
+# Linux Capabilities in Container Runtime Interface
+
+The following proposal summarizes the current state and problems for runtimes to implement Linux Capabilities.
+
+## Problems
+
+Today's [Linux Capabilities Interface](https://github.com/kubernetes/kubernetes/blob/v1.5.0-alpha.0/pkg/kubelet/api/v1alpha1/runtime/api.proto#L350-L355) in CRI is derived from the Docker runtime implementation, which also maps
+to the [k8s API](https://github.com/kubernetes/kubernetes/blob/v1.5.0-alpha.0/pkg/api/v1/types.go#L1164-L1170).
+
+If users want to launch a container with ad-hoc capabilities, they will go through the following steps:
+
+- Get a list of capabilities he would require
+- Look up Docker's docs about the [default capability list](https://docs.docker.com/engine/reference/run/#/runtime-privilege-and-linux-capabilities).
+- Calculate the `Add` and `Drop` capability set based on his requirements and the default capabilities list.
+- Specify the `Add` and `Drop` set for the containers in a k8s pod spec.
+- Launch the pod/container.
+
+There are several issues here:
+
+- The default capability set heavily depends on Docker's implementation, Kubernetes has no control over that.
+When Docker changes the default capability set across versions, the effective set of default capabilities can change suddenly.
+- To implement the same semantics as today's Docker runtime, other runtimes (e.g. rkt) are required to depend on the implicit Docker default capability set as well. And if Docker changes the default capability set, then consistency is not guaranteed.
+- From a security point of view, an explicit capability set which does not rely on undefined defaults is preferable for auditing purposes.
+
+## Solution
+
+The proposed solution will:
+- Keep the k8s API as is today. Users will not need to change anything.
+- Define a default capability whitelist inside k8s, and document it.
+- Add a `DefaultCapabilities` field to the `PodSecurityPolicySpec` which overrides the default capability whitelist inside k8s.
+- Set the default capability whitelist in the `PodSpec`'s `SecurityContext`.
```

Comment:
Ah sorry, one step back and a question: I added the default whitelist to the `PodSpec` due to the following comment https://github.com/kubernetes/kubernetes/pull/33614#discussion_r80986805:

>  When feasible, I think it's better to keep this in spec, rather than status. To do that, we would need to have a capabilities whitelist field on a pod (in addition to add/drop), and resolve the whitelist in the apiserver based on the PSP (unless it's explicitly set by the user, of course).

So if we set the resolved (computed) whitelist in the `PodSpec` instead, it means that a pod author could also freely adjust this to his liking too. It seems strange to me that I can specify a "resolved" whitelist as a pod author at all.

Please advice if I am completely misunderstanding your review comment for the previous proposal or I have addressed it completely wrong ;-)